Jacobi/Sensei/C/solution/JacobiDataAdaptor.cxx

```diff
@@ -252,17 +306,47 @@ vtkDataObject* JacobiDataAdaptor::GetMesh(bool vtkNotUsed(structure_only))
     int n_ranks = 1;
     MPI_Comm_size(MPI_COMM_WORLD, &n_ranks);
 
+#ifdef RECTILINEAR_MESH
+    vtkRectilinearGrid *block = vtkRectilinearGrid::New();
+    int dims[3];
+    dims[0] = this->bx+2;
+    dims[1] = this->by+2;
+    dims[2] = 1;
+    block->SetDimensions(dims);
+
+    vtkFloatArray *x = vtkFloatArray::New();
+    vtkFloatArray *y = vtkFloatArray::New();
+    vtkFloatArray *z = vtkFloatArray::New();
+    int dontDelete = 1;
+    x->SetArray(this->cx, this->bx+2, dontDelete);
+    y->SetArray(this->cy, this->by+2, dontDelete);
+    z->SetNumberOfTuples(1);
+    z->SetTuple1(0,0.);
+    block->SetXCoordinates(x);
+    block->SetYCoordinates(y);
+    block->SetZCoordinates(z);
+    x->Delete();
+    y->Delete();
+    z->Delete();
+    this->Mesh = vtkSmartPointer<vtkMultiBlockDataSet>::New();
+    this->Mesh->SetNumberOfBlocks(1);
```

Comment:
should be
this->Mesh->SetNumberOfBlocks(n_ranks)
this->Mesh->SetBlock(rank, block)